### ISSUE TITLE
bug(cdk/menu): Mac OS VoiceOver cannot operate menu

### ISSUE THREAD

**Issue by oajulianclementson** [angular/components#27376]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

14

### Description

Using VoiceOver on Mac OS, I am unable to open a CDK-enabled menu using the standard VoiceOver key combination for a button (Ctrl-Option-Space).

The same applies to selecting menu items after the menu is open (if opened some other way first).

### Reproduction

Go to the CDK menu examples page: https://material.angular.io/cdk/menu/examples using Mac OS, with VoiceOver enabled. Tab to the button under "Google Docs Menu Bar" or "Menu with Standalone Trigger" or "Stateful Menu with Standalone Trigger".

VoiceOver announces "To display a list of options, press Ctrl-Option-Space". Press this key combination.

### Expected Behavior

Pressing Ctrl-Option-Space should open the menu. This is the standard VoiceOver command for activating any button or link.

### Actual Behavior

Pressing Ctrl-Option-Space has no effect.

The same occurs for v15: https://v15.material.angular.io/cdk/menu/examples

But it works as expected for v14: https://v14.material.angular.io/cdk/menu/examples

It also works as expected for the latest Material menu: https://material.angular.io/components/menu/examples rather than the CDK.

### Environment

- Angular: 15, 16
- CDK: 15, 16
- Browser(s): latest Chrome, Edge, Firefox, or Safari
- Operating System: Mac OS (reproduced on both latest 13.4.1 and the much older 10.13.6)

**Comment by oajulianclementson** [comment]
After conducting some tests overriding these two methods, I now partly understand the reason for the above conditional logic, at least in `CdkMenuTrigger`.

When selecting a menu item using the keyboard with space or enter (not a screen-reader), as well as a keyboard event being received by `CdkMenuItem`, a click event is received by `CdkMenuTrigger`. Without the conditional logic in place, this immediately re-opens the menu.

It seems that `CdkMenuTrigger` needs a way to distinguish between two different scenarios:
1. keyboard event (space or enter) on menu item, followed immediately by click event on trigger: this is the user selecting a menu item using the keyboard. The click should be suppressed, so as not to re-open the menu.
2. any other keyboard event (e.g. arrow keys navigating to the trigger), followed by click event on trigger: this is a user using a screen-reader to open the menu, and the click event should be honoured. This is because the screen-reader issues a click event to open the menu, not a keyboard event.

The current use of `InputModalityDetector` isn't enough to distinguish between these scenarios. We would have to keep track of what the previous keyboard event was, not just the modality.

There are possibly some other edge cases I haven't considered, that would complicate this further.

So far in my testing I haven't found any need for the same conditional logic in `CdkMenuItem` but I may have missed it.
